contracts/modules/GoldenEggMetadata.sol

```diff
@@ -2,38 +2,83 @@
 pragma solidity ^0.8.16;
 
 import { LibString } from "solady/utils/LibString.sol";
+import { OwnableRoles } from "solady/auth/OwnableRoles.sol";
 import { ISoundEditionV1 } from "@core/interfaces/ISoundEditionV1.sol";
 import { IGoldenEggMetadata } from "@modules/interfaces/IGoldenEggMetadata.sol";
 
 contract GoldenEggMetadata is IGoldenEggMetadata {
+    // =============================================================
+    //                            STORAGE
+    // =============================================================
+
+    /**
+     * @dev The maximum `tokenId` for `edition` that has a numbered json.
+     * If zero, all `tokenId`s have number jsons.
+     */
+    mapping(address => uint256) public numberedUpto;
+
+    // =============================================================
+    //               PUBLIC / EXTERNAL WRITE FUNCTIONS
+    // =============================================================
+
+    /**
+     * @inheritdoc IGoldenEggMetadata
+     */
+    function setNumberedUpto(address edition, uint256 tokenId) external onlyEditionOwnerOrAdmin(edition) {
+        numberedUpto[edition] = tokenId;
+        emit NumberUptoSet(edition, tokenId);
+    }
+
+    // =============================================================
+    //               PUBLIC / EXTERNAL VIEW FUNCTIONS
+    // =============================================================
+
     /**
      * @inheritdoc IGoldenEggMetadata
      */
     function tokenURI(uint256 tokenId) external view returns (string memory) {
-        uint256 goldenEggTokenId = getGoldenEggTokenId(ISoundEditionV1(msg.sender));
+        uint256 goldenEggTokenId = getGoldenEggTokenId(msg.sender);
         string memory baseURI = ISoundEditionV1(msg.sender).baseURI();
 
-        if (tokenId == goldenEggTokenId) {
-            return bytes(baseURI).length != 0 ? string.concat(baseURI, "goldenEgg") : "";
-        }
+        if (bytes(baseURI).length == 0) return "";
```

Comment:
e.g. thoughts on falling back to default value of 1000 if not set?

Reply:
sgtm